[Installation]: TypeError: unsupported operand type(s) for |: 'type' and 'NoneType' [ERROR] 2025-10-24-11:31:36 (PID:1609325, Device:-1, RankID:-1) ERR99999 UNKNOWN applicaiton exception

### Your current environment

When excute  "_pip install vllm-ascend==0.11.0rc0_"  , Prompt Error:
_TypeError: unsupported operand type(s) for |: 'type' and 'NoneType'
[ERROR] 2025-10-24-11:31:36 (PID:1609325, Device:-1, RankID:-1) ERR99999 UNKNOWN applicaiton exception_

I checked my current Python version ：
_[root@devserver-314b-0 ~]# python --version
Python 3.9.9_

while The | operator in type annotations requires Python 3.10+.

In our README describe:
Python >= 3.9, < 3.12  ,maybe it's not proper ; ---->suggest "Python >= 3.10, < 3.12 "   I have tested Python3.10,  thta's ok !

--- a/README.md
+++ b/README.md
@@ -41,7 +41,7 @@ By using vLLM Ascend plugin, popular open-source models, including Transformer-l
 - Hardware: Atlas 800I A2 Inference series, Atlas A2 Training series, Atlas 800I A3 Inference series, Atlas A3 Training series, Atlas 300I Duo (Experimental)
 - OS: Linux
 - Software:
-  * Python >= 3.9, < 3.12
+  * Python >= 3.10, < 3.12
   * CANN >= 8.3.rc1 (Ascend HDK version refers to [here](https://www.hiascend.com/document/detail/zh/canncommercial/83RC1/releasenote/releasenote_0000.html))
   * PyTorch == 2.7.1, torch-npu == 2.7.1
   * vLLM (the same version as vllm-ascend)

--- a/README.zh.md
+++ b/README.zh.md
@@ -42,7 +42,7 @@ vLLM 昇腾插件 (`vllm-ascend`) 是一个由社区维护的让vLLM在Ascend NP
 - 硬件：Atlas 800I A2 Inference系列、Atlas A2 Training系列、Atlas 800I A3 Inference系列、Atlas A3 Training系列、Atlas 300I Duo（实验性支持）
 - 操作系统：Linux
 - 软件：
-  * Python >= 3.9, < 3.12
+  * Python >= 3.10, < 3.12
   * CANN >= 8.3.rc1 (Ascend HDK 版本参考[这里](https://www.hiascend.com/document/detail/zh/canncommercial/83RC1/releasenote/releasenote_0000.html))
   * PyTorch == 2.7.1, torch-npu == 2.7.1
   * vLLM (与vllm-ascend版本一致)

--- a/docs/source/installation.md
+++ b/docs/source/installation.md
@@ -5,7 +5,7 @@ This document describes how to install vllm-ascend manually.
 ## Requirements
 
 - OS: Linux
-- Python: >= 3.9, < 3.12
+- Python: >= 3.10, < 3.12
 - A hardware with Ascend NPU. It's usually the Atlas 800 A2 series.
 - Software:
 

--- a/examples/disaggregated_prefill_v1/mooncake_connector_deployment_guide.md
+++ b/examples/disaggregated_prefill_v1/mooncake_connector_deployment_guide.md
@@ -3,7 +3,7 @@
 ## Environmental Dependencies
 
  *  Software:
-     *  Python >= 3.9, < 3.12
+     *  Python >= 3.10, < 3.12
      *  CANN >= 8.3.rc1
      *  PyTorch == 2.7.1, torch-npu == 2.7.1
      *  vLLM (same version as vllm-ascend)

--- a/setup.py
+++ b/setup.py
@@ -373,7 +373,6 @@ setup(
     },
     # TODO: Add 3.12 back when torch-npu support 3.12
     classifiers=[
-        "Programming Language :: Python :: 3.9",
         "Programming Language :: Python :: 3.10",
         "Programming Language :: Python :: 3.11",
         "License :: OSI Approved :: Apache Software License",
@@ -384,7 +383,7 @@ setup(
         "Topic :: Scientific/Engineering :: Information Analysis",
     ],
     packages=find_packages(exclude=("docs", "examples", "tests*", "csrc")),
-    python_requires=">=3.9",
+    python_requires=">=3.10",
     install_requires=get_requirements(),
     ext_modules=ext_modules,
     cmdclass=cmdclass,

